sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -1066,10 +1074,9 @@ private[joins] object LongHashedRelation {
 
 /**
  * Common trait with dummy implementation for NAAJ special HashedRelation
- * EmptyHashedRelation
  * HashedRelationWithAllNullKeys
  */
-trait NullAwareHashedRelation extends HashedRelation with Externalizable {
+trait NullAwareHashedRelation extends HashedRelation {
```

Comment:
we don't need the trait if it only has one sub-class.

Reply:
OK, consider it has no other extension in future NAAJ scenario, it is reasonable to just delete it.